tests/test_juju_topology.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+import unittest
+from collections import OrderedDict
+
+from charms.observability_libs.v0.juju_topology import JujuTopology
+from ops.charm import CharmBase
+from ops.testing import Harness
+
+
+class JujuTopologyCharm(CharmBase):
+    pass
+
+
+class TestJujuTopologyLib(unittest.TestCase):
+    def setUp(self):
+        self.input = OrderedDict(
+            [
+                ("model", "some-model"),
+                ("model_uuid", "00000000-uuid-0000-000000000000"),
+                ("application", "test-application"),
+                ("unit", "test-application/0"),
+                ("charm_name", "test-application"),
+            ]
+        )
+        self.topology = JujuTopology(
+            self.input["model"],
+            self.input["model_uuid"],
+            self.input["application"],
+            self.input["unit"],
+            self.input["charm_name"],
+        )
+
+    def test_dict_includes_all_expected_keys(self):
+        assert self.topology.as_dict() == self.input
```

Comment:
Pretty much all of these should be replaced with `unittest` assertions.

For example:
```python
self.assertEqual(self.topology.as_dict(), self.input)
```

It gives nicer results if the test fails, follows common practices, and gets you a bunch of stuff "free", like `assertContains`

Plus, using these is a good assurance that the cleanup steps (for the harness, for example, if it's ever used here) actually run.